### PR TITLE
fix: Modal's open animation

### LINKED ISSUE #15795: Modal's open animation doesn't start at the click point when rendering for the first time in Table
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/s/x2l04pz46z)

### Steps to reproduce
1. create a Table and add a modal function in each row
2. after the first render, trigger the modal function, Modal will transition from the middle of the page instead of from the click point, and after the first trigger, Modal will open with a normal transition

### What is expected?
`Modal` transition from the click point every time

### What is actually happening?
`Modal` transition from the click point after first trigger

| Environment | Info |
|---|---|
| antd | 3.16.1 |
| React | 16.8.3 |
| System | MacOS 10.14.3 |
| Browser | Chrome 75 |

### PR DESCRIPTION
**Pull request by ztplz**
### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### 👻 What's the background?

fix #15795 

### 📝 Changelog

- English Changelog:
Fix Modal's open animation doesn't start at the click point when rendering for the first time
- Chinese Changelog (optional):

### ☑️ Self Check before Merge

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

**Comment by ztplz** [comment]
https://github.com/ant-design/ant-design/blob/master/components/modal/Modal.tsx#L166
这个问题的原因是点击事件是在didmount里注册的  所以第一次点击的时候 mousePosition是null的  所以我把它移出来了

File: components/modal/Modal.tsx [hunk 1]
```diff
@@ -158,22 +172,8 @@ export default class Modal extends React.Component<ModalProps, {}> {
...
-        y: e.pageY,
-      };
-      // 100ms 内发生过点击事件，则从点击位置动画展示
-      // 否则直接 zoom 展示
-      // 这样可以兼容非点击方式展开
-      setTimeout(() => (mousePosition = null), 100);
-    });
-    mousePositionEventBinded = true;
+  componentWillUnmount() {
+    window.removeEventListener('scroll', getClickPosition);
```
Comment (zombieJ) [inline components/modal/Modal.tsx]: 为什么要移除呢？

Comment (ztplz) [reply components/modal/Modal.tsx]: 手抖了  写错了 click 写成 scroll 了

**Comment by ztplz** [comment]
感觉要换个方法   放外面ssr会挂

**Comment by zombieJ** [comment]
判断一下 window 就行了，另外添加了事件就不用移除了，否则第一个 modal 删除你之后开始点就不更新了。

File: components/modal/Modal.tsx [hunk 2]
```diff
@@ -11,9 +11,25 @@ import LocaleReceiver from '../locale-provider/LocaleReceiver';
...
+    y: e.pageY,
+  };
+  // 100ms 内发生过点击事件，则从点击位置动画展示
+  // 否则直接 zoom 展示
+  // 这样可以兼容非点击方式展开
+  setTimeout(() => (mousePosition = null), 100);
+};
+
+// 只有点击事件支持从鼠标位置动画展开
+if (document) {
```
Comment (afc163) [inline components/modal/Modal.tsx]: 这个判断会有 SSR 的问题。

**Comment by afc163** [comment]
This pull request **introduces 1 alert** when merging 0fc3201d3b8d6a59b63b826d1bc74086732c8730 into 6e2c2baaae553fc1dffab37ca999d11fe11d3f40 - [view on LGTM.com](https://lgtm.com/projects/g/ant-design/ant-design/rev/pr-aed9f261538fcd4407344f994f2464210e015a89)

**new alerts:**

* 1 for Comparison between inconvertible types

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by afc163** [comment]
This pull request **introduces 1 alert** when merging 519a53e3d407168c64fde62076154df5cd5a3020 into 6a754eb32059df217adc0bec0b75a8aeff83bdde - [view on LGTM.com](https://lgtm.com/projects/g/ant-design/ant-design/rev/pr-0e2ce0857bd7566a845cd7dc08c2401d33111bb0)

**new alerts:**

* 1 for Comparison between inconvertible types

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by zombieJ** [comment]
https://github.com/ant-design/ant-design/blob/master/components/_util/styleChecker.tsx#L2
```jsx
if (typeof window !== 'undefined' && window.document && window.document.documentElement) {
```

File: components/modal/Modal.tsx [hunk 3]
```diff
@@ -26,7 +26,7 @@ const getClickPosition = (e: MouseEvent) => {
 };
 
 // 只有点击事件支持从鼠标位置动画展开
-if (typeof window !== undefined) {
+if (typeof window !== undefined && document) {
```
Comment (afc163) [inline components/modal/Modal.tsx]: typeof window !== undefined && typeof document !== undefined

**Comment by ourfeel** [comment]
不移除 带来 微前端加载，  大量的相同事件注册

**Pull request merged**